dataland-frontend/src/components/forms/CreateCompany.vue

```diff
@@ -1,130 +1,382 @@
-      <template v-if="postCompanyProcessed">
-        <SuccessUpload
-          v-if="postCompanyResponse"
-          msg="company"
-          :messageCount="messageCount"
-          :data="postCompanyResponse.data"
-        />
-        <FailedUpload v-else msg="Company" :messageCount="messageCount" />
-      </template>
+        <template v-if="postCompanyProcessed">
+          <SuccessUpload v-if="uploadSucceded" :message="message" :messageId="messageCounter" />
+          <FailedUpload v-else :message="message" :messageId="messageCounter" />
+        </template>
+      </div>
     </template>
   </Card>
 </template>
 
 <script lang="ts">
-import { FormKit, FormKitSchema } from "@formkit/vue";
-import SuccessUpload from "@/components/messages/SuccessUpload.vue";
-import { SchemaGenerator } from "@/services/SchemaGenerator";
-import { ApiClientProvider } from "@/services/ApiClients";
-import backend from "@clients/backend/backendOpenApi.json";
-import FailedUpload from "@/components/messages/FailedUpload.vue";
+import { FormKit } from "@formkit/vue";
 import Card from "primevue/card";
-import PrimeButton from "primevue/button";
 import { defineComponent, inject } from "vue";
 import Keycloak from "keycloak-js";
-import { CompanyInformation } from "@clients/backend";
+import {
+  CompanyInformation,
+  CompanyIdentifier,
+  CompanyIdentifierIdentifierTypeEnum,
+  StoredCompany,
+} from "@clients/backend";
+import { ApiClientProvider } from "@/services/ApiClients";
+import PrimeButton from "primevue/button";
+import { getAllCountryCodes } from "@/utils/CountryCodeConverter";
 import { assertDefined } from "@/utils/TypeScriptUtils";
-
-const companyInformation = backend.components.schemas.CompanyInformation;
-const companyIdentifier = backend.components.schemas.CompanyIdentifier;
-const companyInformationSchemaGenerator = new SchemaGenerator(companyInformation, ["isTeaserCompany"]);
-const companyIdentifierSchemaGenerator = new SchemaGenerator(companyIdentifier);
+import SuccessUpload from "@/components/messages/SuccessUpload.vue";
+import FailedUpload from "@/components/messages/FailedUpload.vue";
+import Tooltip from "primevue/tooltip";
+import {
+  companyDataNames,
+  companyDataExplanations,
+  gicsSectors,
+} from "@/components/resources/frameworkDataSearch/ReferenceDataModelTranslations";
+import UploadFormHeader from "@/components/forms/parts/UploadFormHeader.vue";
+import { AxiosError } from "axios";
+import { FormKitNode } from "@formkit/core";
 
 export default defineComponent({
   name: "CreateCompany",
   components: {
-    FailedUpload,
+    UploadFormHeader,
     Card,
-    PrimeButton,
     FormKit,
-    FormKitSchema,
+    PrimeButton,
     SuccessUpload,
+    FailedUpload,
+  },
+  directives: {
+    tooltip: Tooltip,
   },
+  emits: ["companyCreated"],
   setup() {
     return {
       getKeycloakPromise: inject<() => Promise<Keycloak>>("getKeycloakPromise"),
     };
   },
-
   data: () => ({
+    companyName: "",
```

Comment:
Before this branch, these entries were part of some model to store companyinformation, right? What is the reason that this grouping was removed?

Reply:
Before this change we had a bit of code that autogenerated the upload form from the data model. However, it was rather confusing in the implementation and not very maintainable. Hence, we decided to get rid of it and implement it directly. The values like companyName you see here contain what the user enters on the website and are cast into the companyInformation object as soon as the submit button is pressed.